server2/Arcadia.Assistant/Arcadia.Assistant.Sharepoint/Sharepoint.cs

```diff
@@ -0,0 +1,307 @@
+                        try
+                        {
+                            var removeStorageItems = employeeWorkoutsStorageItems.Where(x => !employeeWorkoutsValues.Keys.Contains(x.CalendarEventId)).ToArray();
+
+                            // insert or update items
+                            foreach (var workHourEventId in employeeWorkoutsValues.Keys)
+                            {
+                                var employeeMetadata = departmentEmployes.Single(x => x.EmployeeId == employeeWorkoutsValues[workHourEventId].EmployeeId);
+                                await this.UpsertWorkHour(workHourEventId, calendar, employeeWorkoutsValues[workHourEventId], employeeMetadata, externalStorage, cancellationToken);
+                            }
+
+                            // remove redundant items
+                            foreach (var item in removeStorageItems)
+                            {
+                                await externalStorage.DeleteItem(
+                                    calendar,
+                                    item.Id);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            ServiceEventSource.Current.ServiceMessage(this.Context, e.ToString());
+                        }
+
+                        #endregion
+
+                        #region sick lives synchronization
+
+                        // synchronize vacations for selected department
+                        try
+                        {
+                            var removeStorageItems = employeeSickLeavesStorageItems.Where(x => !employeeSickLeavesValues.Keys.Contains(x.CalendarEventId)).ToArray();
+
+                            // insert or update items
+                            foreach (var sickLeaveEventId in employeeSickLeavesValues.Keys)
+                            {
+                                var employeeMetadata = departmentEmployes.Single(x => x.EmployeeId == employeeSickLeavesValues[sickLeaveEventId].EmployeeId);
+                                await this.UpsertSickLeave(sickLeaveEventId, calendar, employeeSickLeavesValues[sickLeaveEventId], employeeMetadata, externalStorage, cancellationToken);
+                            }
+
+                            // remove redundant items
+                            foreach (var item in removeStorageItems)
+                            {
+                                await externalStorage.DeleteItem(
+                                    calendar,
+                                    item.Id);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            ServiceEventSource.Current.ServiceMessage(this.Context, e.ToString());
+                        }
+
+                        #endregion
+                    }
+                }
+
+#if DEBUG
```

Comment:
get rid of #if. You should have local changes to appsettings instead

Reply:
Changed